Official 1.14 Release Docs (#13174)
kubernetes/website: content/en/docs/setup/independent/install-kubeadm.md

open. Since this differs with each pod network plugin, please see the
documentation for the plugins about what port(s) those need.

## Installing runtime {#installing-runtime}

Since v1.6.0, Kubernetes has enabled the use of CRI, Container Runtime Interface, by default.

Since v1.14.0, kubeadm will try to automatically detect the container runtime on Linux nodes
by scanning through a list of well known domain sockets. The detectable runtimes and the
socket paths, that are used, can be found in the table below.

| Runtime    | Domain Socket                    |
|------------|----------------------------------|
| Docker     | /var/run/docker.sock             |
| containerd | /run/containerd/containerd.sock  |
| CRI-O      | /var/run/crio/crio.sock          |

If both Docker and containerd are detected together, Docker takes precedence. This is
needed, because Docker 18.09 ships with containerd and both are detectable.
If any other two or more runtimes are detected, kubeadm will exit with an appropriate
error message.

> On non-Linux nodes the container runtime used by default is Docker.

Reviewer: ```suggestion
On non-Linux nodes, kubeadm defaults to using  Docker as the container runtime.
```

(My suggestion, however original wording is OK too).

> If the container runtime of choice is Docker, it is used through the built-in

Reviewer: ```suggestion
When using Docker as the container runtime, Kuberenetes relies on a built-in
```
